pyaedt/modules/AdvancedPostProcessing.py

```diff
@@ -81,179 +89,296 @@ def is_float(istring):
-         show_grid : bool, optional
-             Whether to show the grid. The default is ``True``.
-         show_ruler : bool, optional
-             Whether to show the ruler. The default is ``True``.
+    def __init__(
+        self,
+        path,
+        log_scale=True,
+        coordinate_units="meter",
+        opacity=1,
+        color_map="rainbow",
+        label="Field",
+        tol=1e-3,
+        headers=2,
+    ):
+        self.path = path
+        self.log_scale = log_scale
+        self.units = coordinate_units
+        self.opacity = opacity
+        self.color_map = color_map
+        self._cached_mesh = None
+        self._cached_polydata = None
+        self.label = label
+        self.name = os.path.splitext(self.path)[0]
+        self.color = (255, 0, 0)
+        self.surface_mapping_tolerance = tol
+        self.header_lines = headers
+
+
+class ModelPlotter(object):
+    """Class that manage the plot data."""
+
+    def __init__(self):
+        self._objects = []
+        self._fields = []
+        self._frames = []
+        self.show_axes = True
+        self.show_legend = True
+        self.show_grid = True
+        self.is_notebook = is_notebook()
+        self.gif_file = None
+        self.legend = True
+        self._background_color = (255, 255, 255)
+        self.off_screen = False
+        self.windows_size = [1024, 768]
+        self.pv = None
+        self.view = "isometric"
+        self.units = "meter"
+        self.frame_per_seconds = 1
+        self._plot_meshes = []
+        self.range_min = None
+        self.range_max = None
+        self.image_file = None
+
+    @property
+    def background_color(self):
+        return self._background_color
+
+    @background_color.setter
+    def background_color(self, value):
+        if isinstance(value, (tuple, list)):
+            self._background_color = value
+        elif value in CSS4_COLORS:
+            h = CSS4_COLORS[value].lstrip("#")
+            self._background_color = tuple(int(h[i : i + 2], 16) for i in (0, 2, 4))
+
+    @property
+    def fields(self):
+        """Field Object.
 
         Returns
         -------
-        :class:`IPython.core.display.Image`
-            Jupyter notebook image.
+        list of :class:`pyaedt.modules.AdvancedPostProcessing.FieldClass`
+        """
+        return self._fields
+
+    @property
+    def frames(self):
+        """Frames list for animation.
 
+        Returns
+        -------
+        list of :class:`pyaedt.modules.AdvancedPostProcessing.FieldClass`
         """
-        file_name = self.export_model_picture(show_axis=show_axis, show_grid=show_grid, show_ruler=show_ruler)
-        return Image(file_name, width=500)
+        return self._frames
 
-    @aedt_exception_handler
-    def get_efields_data(self, setup_sweep_name="", ff_setup="Infinite Sphere1", freq="All"):
-        """Compute Etheta and EPhi.
+    @property
+    def objects(self):
+        """Field Object.
```

Comment:
I agree that my description/suggestion is not of great help for the user neither.
```suggestion
        """List of class objects.
```